### DISCUSSION TITLE
Add 2 factor authentication to login

### DISCUSSION THREAD

**Discussion by mitmelon** [postalserver/postal#1983] [Feature suggestions]
Please can you add 2 factor authentication to login for more advanced security.

Thanks.

**Comment by uhlhosting** [comment]
You can always add yourself auth from server side. Check https://authp.github.io , since latest release is using Caddy server.

**Reply by mitmelon** [reply]
The problem I have with this is that... I don't know Ruby programming language... I only know PHP and Rust.

Do you have an idea for how I can implement this...still

2) Please if I have a server of 8GB RAM with 200GB SSD space can I host postal along side with my website on same server or I must still separate the two.

3) Is it possible to change the mariadb and nginx port so it won't conflict with other applications.

**Reply by uhlhosting** [reply]
@mitmelon Caddy is outside ruby, its in fact in JSON,

Configuration is both dynamic and exportable with [Caddy's API](https://caddyserver.com/docs/api). Although no config files required, you can still use them; most people's favorite way of configuring Caddy is using the [Caddyfile](https://caddyserver.com/docs/caddyfile). The format of the config document takes many forms with [config adapters](https://caddyserver.com/docs/config-adapters), but Caddy's native config language is [JSON](https://caddyserver.com/docs/json/).

**Reply by uhlhosting** [reply]
@mitmelon best would be to have the Postal deployed on its own VPS/Server, so you avoid any conflicts.
